File: readme_examples/js/object.assign.2.js
/**
 * Create an instance of Object, with
 * no prototype.
 */
const fruit = Object.create(null)

/**
 * Create another object, with "fruit"
 * as its prototype.
 */
const pineapple = Object.create(fruit)

/**
 * Merge {sour: true} into "pineapple", and then
 * merge "pineapple" into "fruit".
 */
Object.assign(fruit, pineapple, {sour: true})

console.log(fruit.sour)     // => true
console.log(pineapple.sour) // => true

File: readme_examples/rb/object.assign.2.rb
require_relative "setup"
require "ryo"
require "ryo/core_ext/object"

##
# Create an instance of Object, with
# no prototype.
fruit = Object.create(nil)

##
# Create another object, with "fruit"
# as its prototype.
pineapple = Object.create(fruit)

##
# Merge {sour: true} into "pineapple", and then
# merge "pineapple" into "fruit".
Ryo.assign(fruit, pineapple, {sour: true})

puts fruit.sour # => true
puts pineapple.sour # => true
